kooplearn/nn/functional.py

```diff
@@ -1,9 +1,7 @@
+import torch  # noqa: E402
+from typing import Optional  # noqa: E402
 from kooplearn._src.check_deps import check_torch_deps
-
 check_torch_deps()
-from typing import Optional  # noqa: E402
-
-import torch  # noqa: E402
```

Comment:
Importing `torch` should be done only _after_ `check_torch_deps()` has been called